### PR TITLE
Assignment 5 PR

### PR DESCRIPTION
**Pull request by sebastienbeauvais**
Have most everything working. still need to link graphs but they have all been outputted. Run each SQL statement first as the Python code uses a create table.

### CODE REVIEW

**Comment by sebastienbeauvais** [comment]
**READ BEFORE RUNNING**
- For the code to run properly you must execute the SQL file first. This file creates multiple tables which eventually are joined to create the final output table we do all the python modeling on.
- For the python file you will need to create a folder called "graphs" in the same directory as the file is stored in. Or you can remove the 'graphs/' portion on each line of code that a graph is outputted. Without this folder you will get an error saying "no such directory"
- Model scores are at the bottom of the HTML page. To see all the outputs just open assignment_5.html
- I am aware that there are no cells on the first two tables. For some reason when adding clickable links it 'breaks' the tables and we lose CSS formatting :(
- You will also need to download the my_style.css file if you want to see the proper styling. Not sure what will happen if this is not included but my guess is an error will be thrown

File: homework/assignment_5/baseball_features.sql [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+USE baseball;
+SHOW tables FROM baseball;
+show columns from pitcher_stat;
+show columns from pitcher_counts;
+show columns from pitchersInGame;
+
+#####################################################################
+# HOME TEAM WINS
+#####################################################################
+CREATE OR REPLACE TABLE home_team_wins
```
Comment (cidb123) [inline homework/assignment_5/baseball_features.sql:4]: creating tables takes pretty long, each one a few minutes. Could be quicker just using select statement

File: homework/assignment_5/baseball_features.sql [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        ELSE (tpc.Hit+tpc.Walk+tpc.Hit_By_Pitch)/(tpc.atBat+tpc.Walk+tpc.Hit_By_Pitch+tpc.Sac_Fly)
+    END AS away_obp
+FROM
+    game g
+JOIN team_pitching_counts tpc
+ON tpc.team_id = g.away_team_id
+GROUP BY
+    g.game_id
+ORDER BY
+    g.game_id, g.away_team_id DESC;
```
Comment (cidb123) [inline homework/assignment_5/baseball_features.sql:141]: Not totally sure what this error is but I'm seeing an "Aggregate Related Problems" Issue

File: homework/assignment_5/assignment_5.py [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    brute_force = brute_force.rename(columns={"name_url": "Predictor 2 Bin Plot"})
+
+    ##################################################
+    # MODELS
+    ##################################################
+
+    # Decision tree
+    clf = tree.DecisionTreeClassifier()
+    clf = clf.fit(X_train, y_train)
+    tree.plot_tree(clf)
```
Comment (cidb123) [inline homework/assignment_5/assignment_5.py:342]: Getting a weird error here I think has to do with graph viz, I ended up downloading it to my system and then it worked.

File: homework/assignment_5/assignment_5.py [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+            + heat.to_html()
+            + "\n\n"
+            + html_string.format(
+                table=brute_force.to_html(justify="center", classes="mystyle")
+            )
+            + "\n\n"
+            + html_string.format(
+                table=results_df.to_html(justify="center", classes="mystyle")
+            )
+        )
```
Comment (cidb123) [inline homework/assignment_5/assignment_5.py:454]: File runs, but it doesn't automatically for me. And I noticed you have some stats on features like home team vs away team, but you do have good plots that all have clickable links.

File: homework/assignment_5/assignment_5.py [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+        filled=True,
+        rounded=True,
+        special_characters=True,
+    )
+    graph = graphviz.Source(dot_data)
+    graph.render("decision_tree")
+
+    # making a model pipeline
+    model_pipeline = []
+    model_pipeline.append(LogisticRegression(solver="liblinear"))
```
Comment (cidb123) [inline homework/assignment_5/assignment_5.py:357]: Im not sure if it is needed but I think you're missing the t scores and p values in your output

File: homework/assignment_5/assignment_5.py [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+    # sorting values
+    brute_force = brute_force.sort_values(
+        by=["Weighted Difference of Mean Response"], ascending=False
+    )
+    results_df = results_df.sort_values(by=["Accuracy"], ascending=False)
+
+    # creating clickable links
+    output_table = (
+        output_table.style.set_properties(**{"text-align": "center"})
+        .format({"Linear Regression Plot": make_clickable})
```
Comment (cidb123) [inline homework/assignment_5/assignment_5.py:422]: I think this should be a logistic regression if we're predicting wins

**Pull request merged**